pennylane/devices/default_qubit.py

```diff
@@ -66,6 +66,14 @@ def stopping_condition(op: qml.operation.Operator) -> bool:
         return True
     if op.__class__.__name__[:3] == "Pow" and qml.operation.is_trainable(op):
         return False
+    if (
+        isinstance(op, qml.operation.Operator)
+        and not isinstance(op, SymbolicOp)
+        and op.has_sparse_matrix
+        and (not op.has_matrix)
+        and len(op.wires) >= 3
+    ):
+        return True
```

Comment:
I'd like to propose just updating the next few lines to:
```

    return (
        (isinstance(op, Conditional) and stopping_condition(op.base))
        or isinstance(op, MidMeasureMP)
        or op.has_matrix
        or op.has_sparse_matrix
    )
```

Reply:
True. Will come back to merge the logic blocks after CI passes

Reply:
https://github.com/PennyLaneAI/pennylane/pull/6883/commits/acdf6db05390b1e9fcdc2dc6ce65bdc4ac3008a7